src/oaklib/converters/owl_to_fhir_codesystem.py

```diff
@@ -0,0 +1,233 @@
+    if not load_cache or (load_cache and not os.path.exists(cache_path)):
+        # todo: When performance issue (https://github.com/biopragmatics/bioontologies/issues/7) fixed, can move this up
+        from bioontologies import robot
+        parse_results: robot.ParseResults = robot.convert_to_obograph_local(inpath)
+
+    if save_cache:
+        if not os.path.exists(CACHE_DIR):
+            os.mkdir(CACHE_DIR)
+        with open(cache_path, 'wb') as handle:
+            # It can't be unbound given above logic, but PyCharm isn't smart enough to know that.
+            # noinspection PyUnboundLocalVariable
+            pickle.dump(parse_results, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+    graph = parse_results.graph_document.graphs[0]
+    return graph
+
+
+def convert(inpath: str, outpath: str, owned_prefix_map: Dict[str, str], save_cache=False, load_cache=True) -> Dict:
+    """Convert
+
+    Side effects:
+    - Writes file at `outpath`"""
+    g = _load_obograph(inpath=inpath, save_cache=save_cache, load_cache=load_cache)
+
+
+    # todo: using owned_prefix_map, create a dict of <node_id>:<properties>
+    #  - if node.type != 'CLASS', I should pause at breakpoint and take a look
+    nodes: Dict[str, Dict] = {}
+    for node in g.nodes:
+        _id = node.id
+        for prefix, uri in owned_prefix_map.items():
+            _id = _id.replace(uri, prefix + ':')
+        new_node = {
+            "code": _id,
+            # todo: add 'definition'
+            "display": node.lbl,
+            # TODO: 1. property.synonym. valueCoding? (https://build.fhir.org/datatypes.html#Coding) ||
+            #   extension? (look at sssom-py) (https://hl7.org/fhir/extensibility.html) ||
+            #   codeable concept? (https://build.fhir.org/datatypes.html#CodeableConcept)
+            # TODO: I think I need to update GH issue and get feedback from Shahim & zulip / graham etc. There's no canonical way to do this in FHIR.
+            #  obograph repr:
+            #   "synonyms" : [ {
+            #       "pred" : "hasExactSynonym",
+            #       "val" : "malignant neoplasm of cervix uteri",
+            #       "xrefs" : [ "NCIT:C9311" ]}, ...]
+            #  my idea:
+            #  "property": [{
+            #       "code": "http://www.geneontology.org/formats/oboInOwl#hasExactSynonym",  # or maybe shorter
+            #       # https://build.fhir.org/datatypes.html#Coding
+            #       "valueCoding": {
+            #          "system": "<URI>",
+            #          "code": "<CODE>"
+            #          "display": "malignant neoplasm of cervix uteri"
+            #        }
+            #   }, ... ]
+            "property": []
+        }
+        for syn in node.meta.synonyms:
+            # todo: will each Synonym realistically only have 1 xref? It supports a list. But could it be 0 or >1?
+            for xref in syn.xrefs:
```

Comment:
Note that synonym.xrefs is for the *provenance* of the synonym - it's not the code system of the synonym.

E.g.

```yaml
meta:
  synonyms:
    val: "foo disease type 2"
    xrefs:
      - PMID:1234   ### a publications
      - ORCID:4567-897 ### Nicole's/curators orcid
      - ...
```

Reply:
Ah, thanks for clarifying that. I might have made a mistake here.